kubernetes/ansible/roles/helm-deploy/tasks/main.yml

```diff
@@ -14,9 +14,24 @@
     src: "{{ chart_path }}/values.j2"
     dest: "{{ chart_path }}/values.yaml"
 
+- name: Load role to decrypt keys data and copy to helm charts
+  include_role:
+    name: mount-keys
+  vars:
+    private_key_path: "{{ kong_private_key_path | default ('keys') }}"
+    private_key_sign_start: "{{ kong_private_key_sign_start | default(0) }}"
+    private_key_sign_end: "{{ kong_private_key_sign_end | default(1) }}"
+  when: release_name == "adminutils" and adminutil__jwt_keycount is defined and adminutil__jwt_keycount > 0
+
 - name: helm install
   shell: helm upgrade --install --atomic --timeout 10m {{ release_name }} {{ chart_path }} -n {{namespace}}
 
+- name: Remove the keys directory post helm install
```

Comment:
this task will never get executed if the above installation fails for any reason. This will be good for debugging the secrets. But I don't think helm deployment will fail because of secrets. Do we need to add `final` block ?

Reply:
@rjshrjndrn Deployment will fail if keys are incorrect or in different format. Adminutil will crash loop.

Reply:
oh.. Okay. Then make sense to keep the decrypted files, as it'll help to debug